rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -859,6 +844,103 @@ def train(
             shuffle=False,  # we use custom shuffle inside data generator
         )
 
+    def _no_data_debug_log(self):
+        logger.debug(
+            f"Cannot train '{self.__class__.__name__}'. No data was provided. "
+            f"Skipping training of the classifier."
+        )
+
+    def train_on_chunks(
+        self,
+        data_chunk_files: List[DataChunkFile],
+        config: Optional[RasaNLUModelConfig] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Trains this component using the list of data chunk files.
+
+        Args:
+            data_chunk_files: List of data chunk files.
+            config: The model configuration parameters.
+        """
+        if not data_chunk_files:
+            self._no_data_debug_log()
+            return
+
+        # load one chunk so that we can instantiate the model
+        sample_data_chunk = TrainingDataChunk.load_chunk(data_chunk_files[0].file_path)
+        # we do stratified split of nlu examples, if the first one doesn't contain
+        # nlu examples, then no chunk contains them
+        if not sample_data_chunk.nlu_examples:
+            self._no_data_debug_log()
+            return
+
+        self._label_data = self._create_label_data(
+            self._get_index_label_examples(sample_data_chunk, data_chunk_files)
+        )
+
+        sample_model_data = self._preprocess_train_data(sample_data_chunk)
+
+        data_generator = RasaDataChunkFileGenerator(
+            data_chunk_files,
+            self._load_data_func,
+            batch_size=self.component_config[BATCH_SIZES],
+            epochs=self.component_config[EPOCHS],
+            shuffle=True,
+            random_seed=self.component_config[RANDOM_SEED],
+        )
+
+        self._train_model(sample_model_data, data_generator)
```

Comment:
but we run validation in the end of each epoch, not batch

Reply:
Yes, over here as well `data_generator` contains data for whole epoch and not one batch. Since we don't pass any `validation_data_generator` there won't be a validation run. All I'm saying is that we can possibly create a validation data generator as well when training on chunks and pass it to `_train_model` just like we do for usual training on line 926.

Reply:
I suppose, but then we'll need to use a separate chunk for it. Let's do it in a separate PR, I'll add it to the list